### PR TITLE
docs: use `satisfies` for TypeScript code blocks

### PR DESCRIPTION
**Pull request by leerob**
Yay, TS 4.9!

https://twitter.com/leeerob/status/1563540593003106306

### CODE REVIEW

File: docs/02-app/01-building-your-application/07-configuring/01-typescript.mdx [hunk 1]
```diff
@@ -177,19 +177,21 @@ For [`getStaticProps`](/docs/pages/api-reference/functions/get-static-props), [`
 ```tsx filename="pages/blog/[slug].tsx"
 import { GetStaticProps, GetStaticPaths, GetServerSideProps } from 'next'
 
-export const getStaticProps: GetStaticProps = async (context) => {
+export async function getStaticProps(context) {
   // ...
-}
+} satisfies GetStaticProps
```
Comment (almeidx) [inline docs/02-app/01-building-your-application/07-configuring/01-typescript.mdx:182]: This doesn't work. `satisfies` doesn't support functions (https://github.com/microsoft/TypeScript/issues/51556)

In fact, not even the code in your tweet works. You have to wrap the arrow function in parenthesis, i.e.
```ts
export const getStaticProps = (async (context) => {
  // ...
}) satisfies GetStaticProps
```

Comment (leerob) [reply docs/02-app/01-building-your-application/07-configuring/01-typescript.mdx:182]: Thanks, I hadn't tested this yet 😄 Will update.

**Review by mayank1513 — APPROVED** [review]
Looks good 👌

**Review by balazsorban44 — APPROVED** [review]

File: docs/03-pages/01-building-your-application/03-data-fetching/02-get-static-paths.mdx [hunk 2]
```diff
@@ -30,15 +30,15 @@ export const getStaticPaths: GetStaticPaths = async () => {
     ],
     fallback: true, // false or "blocking"
   }
-}
+}) satisfies getStaticPaths
```
Comment (imranbarbhuiya) [inline docs/03-pages/01-building-your-application/03-data-fetching/02-get-static-paths.mdx:33]:
```suggestion
}) satisfies GetStaticPaths
```

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

File: docs/03-pages/01-building-your-application/03-data-fetching/02-get-static-paths.mdx (hunk 2 above)
Comment (leerob) [reply docs/03-pages/01-building-your-application/03-data-fetching/02-get-static-paths.mdx:33]: Good catch: https://github.com/vercel/next.js/pull/55234

**Pull request merged**